featuremanagement/aio/_featuremanager.py

```diff
@@ -120,106 +116,115 @@ def _variant_name_to_variant(self, feature_flag, variant_name):
         feature_filters = feature_conditions.client_filters
 
         if len(feature_filters) == 0:
             # Feature flags without any filters return evaluate
-            result["enabled"] = True
+            evaluation_event.enabled = True
         else:
             # The assumed value is no filters is based on the requirement type.
             # Requirement type Any assumes false until proven true, All assumes true until proven false
-            result["enabled"] = feature_conditions.requirement_type == REQUIREMENT_TYPE_ALL
+            evaluation_event.enabled = feature_conditions.requirement_type == REQUIREMENT_TYPE_ALL
 
         for feature_filter in feature_filters:
             filter_name = feature_filter[FEATURE_FILTER_NAME]
             if filter_name not in self._filters:
-                raise ValueError(f"Feature flag {feature_flag_id} has unknown filter {filter_name}")
+                raise ValueError(f"Feature flag {feature_flag.name} has unknown filter {filter_name}")
             if feature_conditions.requirement_type == REQUIREMENT_TYPE_ALL:
                 if not await self._filters[filter_name].evaluate(feature_filter, **kwargs):
-                    result["enabled"] = False
+                    evaluation_event.enabled = False
                     break
             else:
                 if await self._filters[filter_name].evaluate(feature_filter, **kwargs):
-                    result["enabled"] = True
+                    evaluation_event.enabled = True
                     break
+        return evaluation_event
 
+    def _assign_allocation(self, feature_flag, evaluation_event, **kwargs):
         if feature_flag.allocation and feature_flag.variants:
-            variant_name = self._assign_variant(feature_flag, **kwargs)
+            default_enabled = evaluation_event.enabled
+            variant_name, evaluation_event = self._assign_variant(feature_flag, **kwargs)
+            evaluation_event.enabled = default_enabled
             if variant_name:
-                result["enabled"] = FeatureManager._check_variant_override(
-                    feature_flag.variants, variant_name, result["enabled"]
+                evaluation_event = FeatureManager._check_variant_override(
+                    feature_flag.variants, variant_name, evaluation_event.enabled
                 )
-                result["variant"] = self._variant_name_to_variant(feature_flag, variant_name)
-                return result
+                evaluation_event.variant = self._variant_name_to_variant(feature_flag, variant_name)
+                evaluation_event.feature = feature_flag
+                return evaluation_event
 
         variant_name = None
-        if result["enabled"]:
-            result["enabled"] = FeatureManager._check_default_enabled_variant(feature_flag)
+        if evaluation_event.enabled:
+            evaluation_event = FeatureManager._check_default_enabled_variant(feature_flag)
```

Comment:
There's a lot of evaluation event re-creation happening. I would expect a single evaluation event to be created at the start and for that evaluation event to be the one returned. I find it hard to follow the code in the current state. Some comments could help as well

Reply:
So, I think I cleared the first part up a bit. `EvaluationEvent` is now only created once we need it. I'll try adding some more comments.